snapcraft/cli/remote.py

```diff
@@ -0,0 +1,245 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2019 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import click
+import os
+import uuid
+
+from snapcraft.internal.remote_build import Worktree, LaunchpadClient, Repo, errors
+from typing import List, Tuple
+from xdg import BaseDirectory
+from . import echo
+from ._options import get_project
+from snapcraft import InfoFile
+
+_SUPPORTED_ARCHS = ["amd64", "arm64", "armhf", "i386", "ppc64el", "s390x"]
+
+
+@click.group()
+def remotecli():
+    """Remote build commands"""
+    pass
+
+
+@remotecli.command("remote-build")
+@click.option(
+    "--recover",
+    metavar="<build-number>",
+    type=int,
+    nargs=1,
+    required=False,
+    help="Recover interrupted remote build.",
+)
+@click.option(
+    "--status",
+    metavar="<build-number>",
+    type=int,
+    nargs=1,
+    required=False,
+    help="Display remote build status.",
+)
+@click.option(
+    "--arch",
+    metavar="<arch-list>",
+    type=str,
+    nargs=1,
+    required=False,
+    help="Set architectures to build.",
+)
+@click.option(
+    "--git", is_flag=True, required=False, help="Build a local git repository."
+)
+@click.option(
+    "--user", metavar="<username>", nargs=1, required=False, help="Launchpad username."
+)
+def remote_build(
+    recover: int, status: int, user: str, arch: str, git: bool, echoer=echo
+) -> None:
+    """Dispatch a snap for remote build.
+
+    Command remote-build sends the current project to be built remotely. After the build
+    is complete, packages for each architecture are retrieved and will be available in
+    the local filesystem.
+
+    If the local files are under git revision control, use option --git to build the
+    current branch of this repository. Be aware, however, that untracked files and
+    uncommitted changes will not be used.
+
+    If not specified in the snapcraft.yaml file, the list of architectures to build
```

Comment:
Feels like it should be the other way around, right?
CLI params like --arch should override what is pre-defined, don't you think?

Reply:
Agree that parameters should override, but the precedence is set by the remote builder in this case. As discussed in our meeting, will refactor to not allow both set at the same time.